lightning/src/ln/msgs.rs

```diff
@@ -1682,6 +1682,19 @@ impl Writeable for ReplyShortChannelIdsEnd {
 	}
 }
 
+impl QueryChannelRange {
+	/**
+	 * Calculates the overflow safe ending block height for the query.
+	 * Overflow returns `0xffffffff`, otherwise returns `first_blocknum + number_of_blocks`
+	 */
+	pub fn end_blocknum(&self) -> u32 {
+		match self.first_blocknum.checked_add(self.number_of_blocks) {
+			Some(block) => block,
+			None => 0xffffffff,
```

Comment:
Added constants to scid_utils, however in this spot, we need uint32 max instead of the `0xff_ff_ff` constant in the scid calcs to enable reply sequencing.

Reply:
Ah, in that case best to call `std::u32::max_value()` otherwise I won't get confused :)